tfe/resource_tfe_no_code_module.go

```diff
@@ -0,0 +1,226 @@
+							ForceNew: false,
+							Required: true,
+							MinItems: 1,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceTFENoCodeModuleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(ConfiguredClient)
+
+	options := tfe.RegistryNoCodeModuleCreateOptions{
+		FollowLatestVersion: tfe.Bool(d.Get("follow_latest_version").(bool)),
+		Enabled:             tfe.Bool(d.Get("enabled").(bool)),
+		RegistryModule: &tfe.RegistryModule{
+			ID: d.Get("registry_module").(string),
+		},
+	}
+
+	if variableOptions, ok := d.GetOk("variable_options"); ok {
+		options.VariableOptions = variableOptionsMaptoStruct(variableOptions.([]interface{}))
+	}
+	if versionPin, ok := d.GetOk("version_pin"); ok {
+		options.VersionPin = tfe.String(versionPin.(string))
+	}
+
+	orgName := d.Get("organization").(string)
+
+	log.Printf("[DEBUG] Create no-code module for registry module %s", options.RegistryModule.ID)
+	noCodeModule, err := config.Client.RegistryNoCodeModules.Create(ctx, orgName, options)
+
+	if err != nil {
+		return diag.Errorf("Error creating no-code module for registry module %s: %s", options.RegistryModule.ID, err)
+	}
+
+	d.SetId(noCodeModule.ID)
+	return resourceTFENoCodeModuleRead(ctx, d, meta)
+}
+
+func variableOptionsMaptoStruct(variableOptions []interface{}) []*tfe.NoCodeVariableOption {
+	var variableOptionsRes []*tfe.NoCodeVariableOption
+	for _, v := range variableOptions {
+		vOpt := v.(map[string]interface{})
+		option := &tfe.NoCodeVariableOption{
+			VariableName: vOpt["name"].(string),
+			VariableType: vOpt["type"].(string),
+		}
+		if vOpt["options"] != nil {
+			for _, o := range vOpt["options"].([]interface{}) {
+				option.Options = append(option.Options, o.(string))
+			}
+		}
+		variableOptionsRes = append(variableOptionsRes, option)
+	}
+	return variableOptionsRes
+}
+
+func resourceTFENoCodeModuleUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(ConfiguredClient)
+
+	readOpts := &tfe.RegistryNoCodeModuleReadOptions{
+		Include: []tfe.RegistryNoCodeModuleIncludeOpt{tfe.RegistryNoCodeIncludeVariableOptions},
+	}
+	noCodeModule, err := config.Client.RegistryNoCodeModules.Read(ctx, d.Id(), readOpts)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	options := tfe.RegistryNoCodeModuleUpdateOptions{
+		FollowLatestVersion: tfe.Bool(d.Get("follow_latest_version").(bool)),
+		Enabled:             tfe.Bool(d.Get("enabled").(bool)),
+		RegistryModule:      &tfe.RegistryModule{ID: d.Get("registry_module").(string)},
+	}
+
+	if versionPin, ok := d.GetOk("version_pin"); ok {
+		options.VersionPin = tfe.String(versionPin.(string))
+	}
+	if variableOptions, ok := d.GetOk("variable_options"); ok {
+		options.VariableOptions = variableOptions.([]*tfe.NoCodeVariableOption)
+	}
+
+	err = retry.RetryContext(ctx, time.Duration(5)*time.Minute, func() *retry.RetryError {
```

Comment:
This is also copied over from the registry_module file. But generally, since the Update endpoint is idempotent, a retry will not break the infrastructure.